Library/Homebrew/formula.rb

```diff
@@ -476,6 +476,29 @@ def compiler_failures
     active_spec.compiler_failures
   end
 
+  # The best spec we can upgrade the {Formula} to so it can satisfy dependency.
+  def best_upgrade_spec_sym
```

Comment:
Cool then I think stay on HEAD unless it's removed and then handle as expected. Also, remove the option as it adds a lot of code complexity for a use case I'm not sure we even want to support.